### PR TITLE
jv-spring-rest-web-practice

### PR DESCRIPTION
**Pull request by MiaHrynda**
Created DTOs for users, orders, and shopping carts. Added validation to user DTO. Created all required controllers.

### CODE REVIEW

**Review by Koriastras — APPROVED** [review]

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userId) {
+        return mapper.toDto(orderService
+                .completeOrder(shoppingCartService.getByUser(userService.get(userId))));
+    }
+
+    @GetMapping("/{userId}")
+    public List<OrderResponseDto> getOrdersHistory(@PathVariable Long userId) {
```
Comment (Max-Pochepets) [inline src/main/java/mate/academy/spring/controller/OrderController.java:40]: image

File: src/main/java/mate/academy/spring/model/dto/response/OrderResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package mate.academy.spring.model.dto.response;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsIds;
+    private LocalDateTime orderDate;
```
Comment (Max-Pochepets) [inline src/main/java/mate/academy/spring/model/dto/response/OrderResponseDto.java:9]: let's return in response some simple data types, like primitives, wrappers and String. smth like LocalDateTime will be hard to parse

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]
use single quotes in your commits to avoid tagging real people :)
image
